src/beamlime/constructors/inspectors.py

```diff
@@ -37,26 +37,38 @@ def validate_annotation(annotation: Any) -> Literal[True]:
 
 
 Product = TypeVar("Product")
+
+
+def _is_type_newtype_py310(tp: Any) -> bool:
+    from typing import NewType
+
+    return isinstance(NewType, type) and isinstance(tp, NewType)
+
+
 _NewT = NewType("_NewT", int)
 
 
+def _is_type_newtype_py39(tp: Any) -> bool:
+    return (
+        callable(tp)
+        and hasattr(tp, "__supertype__")
+        and _NewT.__module__ == tp.__module__
+        and _NewT.__qualname__ == tp.__qualname__
+    )
+
+
 class NewTypeMeta(type):
     def __instancecheck__(cls, instance: Any) -> bool:
-        return (
-            callable(instance)
-            and hasattr(instance, "__supertype__")
-            and _NewT.__module__ == instance.__module__
-            and _NewT.__qualname__ == instance.__qualname__
-        )
+        return _is_type_newtype_py310(instance) or _is_type_newtype_py39(instance)
```

Comment:
Shouldn't this use `sys.version < (3, 10)` or something similar to select one of the two functions?

Reply:
That is true... I just added the check